File: javav2/example_code/bedrock-runtime/src/test/java/actions/TestInvokeModel.java
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package actions;

import java.util.stream.Stream;

public class TestInvokeModel extends AbstractModelTest {
    protected Stream<ModelTest> modelProvider() {
        return Stream.of(
                new ModelTest("Claude", com.example.bedrockruntime.models.anthropicClaude.InvokeModel::invokeModel),
                new ModelTest("CohereCommandR", com.example.bedrockruntime.models.cohereCommand.Command_R_InvokeModel::invokeModel),
                new ModelTest("Llama", com.example.bedrockruntime.models.metaLlama.Llama3_InvokeModel::invokeModel),
                new ModelTest("Mistral", com.example.bedrockruntime.models.mistral.InvokeModel::invokeModel)
        );
    }
}

File: kotlin/services/bedrock-runtime/src/test/kotlin/models/TestInvokeModel.kt
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0v

package models

import java.util.stream.Stream

/**
 * Test class for generative AI models on Amazon Bedrock using the InvokeModel API.
 */
class TestInvokeModel : AbstractModelTest() {
    /**
     * Provides test configurations for generative AI models on Amazon Bedrock.
     * Creates test cases that validate each model's ability to generate
     * and return text or byte[] responses.
     */
    override fun modelProvider(): Stream<ModelTest> = listOf(
        ModelTest("Amazon Titan Text") { com.example.bedrockruntime.models.amazon.titan.text.invokeModel() },
        ModelTest("Amazon Nova Canvas") { com.example.bedrockruntime.models.amazon.nova.canvas.invokeModel() },
    ).stream()
}
